torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -201,21 +214,26 @@ void removeProfileNodesAndSpecializeTypes(Block* b) {
       // a use with a different profiled type, we will still be correct.
       // In the future we could consider unifying the types of uses, or adding a
       // type refinement node so uses can have the correct corresponding type.
-      if (profiled_type == TensorType::get()) {
+      if (profiled_type == TensorType::get() && !input_is_optional) {
         continue;
       }
+
+      if (input_is_optional) {
```

Comment:
nit: I wonder if we could pull this before `if (profiled_type == TensorType::get() && !input_is_optional)` so we don't have to check for `!input_is_optional`

This way the intent to ignore profiling information from optional tensors might be a bit clearer.